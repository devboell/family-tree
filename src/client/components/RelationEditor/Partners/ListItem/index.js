import React from 'react'
import { PropTypes } from 'prop-types'

import PartnerItem from './PartnerItem'
import RemoveButton from './RemoveButton'
import ChildrenDiv from './ChildrenDiv'

const ListItem = ({ input: { value }, onRemove }) =>
  <li>
    <PartnerItem>
      <RemoveButton type="button" onClick={onRemove}>X</RemoveButton>
      <p>{value.partner.name}</p>
    </PartnerItem>
    <ChildrenDiv>
      {value.children.map(child => (
        <p
          key={`child-${child.name}`}
        >
          {child.name}
        </p>
      ))}
    </ChildrenDiv>
  </li>

ListItem.propTypes = {
  input: PropTypes.shape({}).isRequired,
  onRemove: PropTypes.func.isRequired,
}

export default ListItem
