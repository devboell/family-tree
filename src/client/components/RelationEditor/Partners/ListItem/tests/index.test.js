import React from 'react'
import ListItem from '../index'


const props = {
  input: {
    value: {
      partner: {
        name: 'John',
      },
      children: [
        {
          name: 'William',
        },
        {
          name: 'Mary',
        },
      ],
    },
  },
  onRemove: jest.fn(),
}

it('ListItem component, renders without crashing', () => {
  expect(shallow(<ListItem {...props} />)).toMatchSnapshot()
})
