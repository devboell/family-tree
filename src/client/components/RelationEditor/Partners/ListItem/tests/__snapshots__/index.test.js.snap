// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component, renders without crashing 1`] = `
<li>
  <styled.div>
    <styled.button
      onClick={[Function]}
      type="button"
    >
      X
    </styled.button>
    <p>
      John
    </p>
  </styled.div>
  <styled.div>
    <p>
      William
    </p>
    <p>
      Mary
    </p>
  </styled.div>
</li>
`;
